Helper/Order.php

```diff
@@ -725,6 +725,158 @@ public function saveUpdateOrder($reference, $boltTraceId = null)
+                $report->setMetaData([
+                    'CREATE ORDER' => [
+                        'pre-auth order.create' => true,
+                        'order increment ID' => $quote->getReservedOrderId(),
+                        'parent quote ID' => $quote->getId(),
+                    ]
+                ]);
+            });
+
+            throw new BoltException(__(
+                'Quote Submit Error. Order #: %1 Parent Quote ID: %2',
+                $quote->getReservedOrderId(),
+                $quote->getId(),
+                null,
+                CreateOrder::E_BOLT_GENERAL_ERROR
+            ));
+        }
+
+        // Check and fix tax mismatch
+        if ($this->configHelper->shouldAdjustTaxMismatch()) {
+            $this->adjustTaxMismatch($transaction, $order, $quote);
+        }
+
+        if (isset($transaction->reference)) {
+            // Save reference to the Bolt transaction with the order
+            $order->addStatusHistoryComment(
+                __('Bolt transaction: %1', $this->formatReferenceUrl($transaction->reference))
+            );
+        }
+
+        if (Hook::$fromBolt) {
+            $order->addStatusHistoryComment(
+                "BOLTPAY INFO :: THIS ORDER WAS CREATED VIA PRE-AUTH WEBHOOK"
+            );
+            // Send order confirmation email to customer.
+            // Emulate frontend area in order for email
+            // template to be loaded from the correct path
+            // even if run from the hook.
+            if (!$order->getEmailSent()) {
+                $this->appState->emulateAreaCode('frontend', function () use ($order) {
+                    $this->emailSender->send($order);
+                });
+            }
+        }
+
+        $order->save();
+
+        return $order;
+    }
+
+    /**
+     * @param Quote $quote
+     * @return bool
+     * @throws LocalizedException
+     */
+    private function isOrderExist($quote)
+    {
+        /** @var OrderModel $order */
+        $order = $this->cartHelper->getOrderByIncrementId($quote->getReservedOrderId());
+
+        if ($order && $order->getId()) {
+            $this->bugsnag->registerCallback(function ($report) use ($quote) {
+                $report->setMetaData([
+                    'Is Order Exist' => [
+                        'pre-auth order.create' => true,
+                        'parent quote ID' => $quote->getId(),
+                        'order reserved ID' => $quote->getReservedOrderId(),
+                    ]
+                ]);
+            });
+            throw new BoltException(
+                __('Duplicate Order Creation Attempt. Order #: %1', $quote->getReservedOrderId()),
+                null,
+                \Bolt\Boltpay\Model\Api\CreateOrder::E_BOLT_ORDER_ALREADY_EXISTS
+            );
+        }
+
+        return false;
+    }
+
+    /**
+     * @param $immutableQuote
+     * @param $transaction
```

Comment:
Can you add types here as you did above?

Reply:
What do you mean?